src/client/eppo-client.ts

```diff
@@ -1267,6 +1279,105 @@ export default class EppoClient {
       sdkLibVersion: LIB_VERSION,
     };
   }
+
+  private getAllBandits(
+    subjectKey: string,
+    subjectAttributes: ContextAttributes,
+    banditActions: Record<string, BanditActions>,
+    flags: Record<string, PrecomputedFlag>,
+  ): Record<string, Record<string, IPrecomputedBandit>> {
+    const banditResults: Record<string, Record<string, IPrecomputedBandit>> = {};
+
+    // Computing Bandits
+    // The first case is easy: a flag resolves to a bandit-key for this subject; compute that bandit.
+    // The second case is more involved: the flag resolves to null. On the client side, the user can now enter a bandit
+    // key as the `default` variation, so we need have every bandit referenced by this flag computed and available.
+    Object.keys(banditActions).forEach((flagKey: string) => {
+      // First, check how the flag evaluated.
+      const flagVariation = flags[flagKey];
+      if (flagVariation) {
+        banditResults[flagKey] ??= {};
+
+        // First case: flag resolved to a value, check if it's a bandit and if so, compute it.
+        const precomputedResult = this.getPrecomputedBandit(
+          flagKey,
+          flagVariation.variationValue,
+          subjectKey,
+          subjectAttributes,
+          banditActions[flagKey],
+        );
+        if (precomputedResult) {
+          banditResults[flagKey][flagVariation.variationValue] = precomputedResult;
+        }
+      } else {
+        // Second case; compute all the bandits referenced by this flag.
+        const banditVariations = this.banditVariationConfigurationStore?.get(flagKey);
+        if (!banditVariations) {
+          return;
+        }
+        banditResults[flagKey] = {};
```

Comment:
minor: same here — this may produce empty maps if bandit evaluation below returns no results. (This make evaluation on the client side slower than it needs to be.)